### PR TITLE
the binary content is not required in the media widget

### PR DESCRIPTION
**Pull request by OskarStark**
Refs #838

you can finde more information about this PR here: 
https://github.com/sonata-project/SonataMediaBundle/issues/838#issuecomment-137134101

### CODE REVIEW

**Comment by rande** [comment]
@OskarStark there is an issue with CS: 

```
Legend: ?-unknown, I-invalid file syntax, file ignored, .-no changes, F-fixed, E-error
   1) Provider/FileProvider.php (multiline_array_trailing_comma)
```

**Comment by OskarStark** [comment]
done @rande 

but i'm not sure if this is correct, is this method used to create a new media file in den media overview?

because there it has to be mandatory...

**Comment by rande** [comment]
Well, I think you can rely on the context information from the form ``$formBuilder->getOption('context') == 'api'`` to fine tune when the binaryContent is mandatory or not. At least with the API, it is required.

**Comment by OskarStark** [comment]
what does api in this case mean?

would you introduce a widget context?

**Comment by rande** [comment]
The api context is used with the API controller. maybe this can be applied to other use case, but I am not a big fan of such solution. Maybe we need to review the form documentation for an alternative solution.

**Comment by OskarStark** [comment]
yes or we can profide a seperate Form Builder like ```buildWidgetForm```

**Comment by OskarStark** [comment]
my fix worked, for creating a new one we use another formbuilder

new:
bildschirmfoto 2015-09-18 um 21 23 27

widget:
bildschirmfoto 2015-09-18 um 21 24 52

**Comment by OskarStark** [comment]
@rande please merge

**Comment by OskarStark** [comment]
@rande can we please merge this?

File: Provider/FileProvider.php [hunk 1]
```diff
@@ -117,7 +117,9 @@ public function buildCreateForm(FormMapper $formMapper)
      */
     public function buildMediaType(FormBuilder $formBuilder)
     {
-        $formBuilder->add('binaryContent', 'file');
+        $formBuilder->add('binaryContent', 'file', array(
+            'required' => false,
```
Comment (rande) [inline Provider/FileProvider.php]: It is mandatory for the api context, so maybe just do ``$formBuilder->getOption('context') == 'api'`` for now

Comment (OskarStark) [inline Provider/FileProvider.php]: done, what do you think?

**Comment by OskarStark** [comment]
ping @Soullivaneuh @rande

**Pull request merged**